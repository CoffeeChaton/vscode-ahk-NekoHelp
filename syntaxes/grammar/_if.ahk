; https://www.autohotkey.com/docs/v1/lib/_If.htm#Basic_Operation

MyWindowTitle := "My Window"
#If WinActive("ahk_class Notepad") or WinActive(MyWindowTitle) ;comment
#Space::MsgBox % "You pressed Win+Spacebar in Notepad or " MyWindowTitle ;comment
~F1::MsgBox % "You pressed Win+Spacebar in Notepad or " MyWindowTitle ;comment
