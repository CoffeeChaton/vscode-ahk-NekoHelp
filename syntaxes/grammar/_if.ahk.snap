>; https://www.autohotkey.com/docs/v1/lib/_If.htm#Basic_Operation
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.ahk meta.comment.line.semicolon.ahk comment.line.semicolon.ahk
>
>MyWindowTitle := "My Window"
#^^^^^^^^^^^^^^ source.ahk
#              ^^ source.ahk keyword.operator.assignment.ahk
#                ^ source.ahk
#                 ^ source.ahk string.quoted.double.ahk punctuation.definition.string.begin.ahk
#                  ^^^^^^^^^ source.ahk string.quoted.double.ahk
#                           ^ source.ahk string.quoted.double.ahk punctuation.definition.string.end.ahk
>#If WinActive("ahk_class Notepad") or WinActive(MyWindowTitle) ;comment
#^^^ source.ahk meta._if.keyword.control.directives.ahk keyword.control.directives._if.ahk
#   ^ source.ahk meta._if.keyword.control.directives.ahk
#    ^^^^^^^^^ source.ahk support.function.ahk
#             ^ source.ahk punctuation.bracket.ahk
#              ^ source.ahk string.quoted.double.ahk punctuation.definition.string.begin.ahk
#               ^^^^^^^^^^^^^^^^^ source.ahk string.quoted.double.ahk
#                                ^ source.ahk string.quoted.double.ahk punctuation.definition.string.end.ahk
#                                 ^ source.ahk punctuation.bracket.ahk
#                                  ^^^^ source.ahk keyword.other.operator.hasdoc.ahk
#                                      ^^^^^^^^^ source.ahk support.function.ahk
#                                               ^ source.ahk punctuation.bracket.ahk
#                                                ^^^^^^^^^^^^^ source.ahk
#                                                             ^ source.ahk punctuation.bracket.ahk
#                                                              ^ source.ahk meta.comment.line.semicolon.ahk
#                                                               ^^^^^^^^ source.ahk meta.comment.line.semicolon.ahk comment.line.semicolon.ahk
>#Space::MsgBox % "You pressed Win+Spacebar in Notepad or " MyWindowTitle ;comment
#^^^^^^ source.ahk hotkey.ahk string.entity.name.function.label.hotkey.ahk
#      ^^ source.ahk hotkey.ahk punctuation.definition.equals.colon.ahk
#        ^^^^^^ source.ahk meta.command.line.ahk support.function.ahk
#              ^ source.ahk meta.command.line.ahk
#               ^ source.ahk meta.command.line.ahk punctuation.definition.variable.percent.ahk
#                ^ source.ahk meta.command.line.ahk
#                 ^ source.ahk meta.command.line.ahk string.quoted.double.ahk punctuation.definition.string.begin.ahk
#                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.ahk meta.command.line.ahk string.quoted.double.ahk
#                                                         ^ source.ahk meta.command.line.ahk string.quoted.double.ahk punctuation.definition.string.end.ahk
#                                                          ^^^^^^^^^^^^^^ source.ahk meta.command.line.ahk
#                                                                        ^ source.ahk meta.command.line.ahk meta.comment.line.semicolon.ahk
#                                                                         ^^^^^^^^ source.ahk meta.command.line.ahk meta.comment.line.semicolon.ahk comment.line.semicolon.ahk
>~F1::MsgBox % "You pressed Win+Spacebar in Notepad or " MyWindowTitle ;comment
#^^^ source.ahk hotkey.ahk string.entity.name.function.label.hotkey.ahk
#   ^^ source.ahk hotkey.ahk punctuation.definition.equals.colon.ahk
#     ^^^^^^ source.ahk meta.command.line.ahk support.function.ahk
#           ^ source.ahk meta.command.line.ahk
#            ^ source.ahk meta.command.line.ahk punctuation.definition.variable.percent.ahk
#             ^ source.ahk meta.command.line.ahk
#              ^ source.ahk meta.command.line.ahk string.quoted.double.ahk punctuation.definition.string.begin.ahk
#               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.ahk meta.command.line.ahk string.quoted.double.ahk
#                                                      ^ source.ahk meta.command.line.ahk string.quoted.double.ahk punctuation.definition.string.end.ahk
#                                                       ^^^^^^^^^^^^^^ source.ahk meta.command.line.ahk
#                                                                     ^ source.ahk meta.command.line.ahk meta.comment.line.semicolon.ahk
#                                                                      ^^^^^^^^ source.ahk meta.command.line.ahk meta.comment.line.semicolon.ahk comment.line.semicolon.ahk
>